File: frontend/src/api/reports.js
// Wraps the Stage 3 /api/reports/* endpoints. These endpoints don't
// exist on the backend yet — see app/schemas/report.py for the exact
// response contract this is built against. Each function fails
// independently so one missing endpoint doesn't break the whole
// Reports page.
import { apiClient } from './client';

export function getWorkforceOverview() {
  return apiClient.get('/api/reports/overview');
}

export function getDepartmentAnalysis() {
  return apiClient.get('/api/reports/departments');
}

export function getSalaryAnalysis() {
  return apiClient.get('/api/reports/salary');
}

export function getJoiningTrends() {
  return apiClient.get('/api/reports/joining-trends');
}

export function getStatusAnalysis() {
  return apiClient.get('/api/reports/status');
}

export function getDataQuality() {
  return apiClient.get('/api/reports/data-quality');
}

File: app/api/reports.py
from fastapi import APIRouter

from app.schemas.report import (
    DataQualityReport,
    DepartmentStats,
    JoiningYearCount,
    SalaryAnalysis,
    StatusCount,
    WorkforceOverview,
)
from app.services import report_service


router = APIRouter(
    prefix="/api/reports",
    tags=["Reports"],
)


@router.get("/overview", response_model=WorkforceOverview)
def workforce_overview():
    return report_service.get_workforce_overview()


@router.get("/departments", response_model=list[DepartmentStats])
def department_analysis():
    return report_service.get_department_analysis()


@router.get("/salary", response_model=SalaryAnalysis)
def salary_analysis():
    return report_service.get_salary_analysis()


@router.get("/joining-trends", response_model=list[JoiningYearCount])
def joining_trends():
    return report_service.get_joining_trends()


@router.get("/status", response_model=list[StatusCount])
def status_analysis():
    return report_service.get_status_analysis()


@router.get("/data-quality", response_model=DataQualityReport)
def data_quality():
    return report_service.get_data_quality()
